### ISSUE TITLE
[no-unused-vars] conflicts with "eslint-plugin-react"

### ISSUE THREAD

**Issue by Kol96** [typescript-eslint/typescript-eslint#446]
**Repro**

```JSON
{
  "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "warn"
  }
}
```

```TSX
// your repro code case
import React from 'react'

export default function() {
  const h1
  return <h1>no-unused-vars</h1>
}
```

**Expected Result**

Lint error

**Actual Result**

No lint error

**Additional Info**

lint will ignore when the jsx element is used as same as the variable is declared.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.6.0` |
| `@typescript-eslint/parser`        | `1.6.0` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.12.0` |
| `eslint-plugin-react`       | `7.12.4` |
| `node`                             | `8.11.1` |
| `npm`                              | `5.10.0` |
